### ISSUE TITLE
Failure to write program (PICKit4)

### ISSUE THREAD

**Issue by krister-ts** [callwyat/mplab-extension#14]
I try to program a target with PICKit4 using the mdb type launch.json config. The device is found by PICKit4 but fails to program.

What works:

- Using toolType "Sim" can debug in vscode
- MPLab X can debug the same elf file

So I am thinking maybe it has to do with toolOptions. I am using the default tool options in MPLab, so I was assuming that the launch.json "toolOptions" field can be left empty. Was I wrong?

Here is my config:
```json
{
    "type": "mdb",
    "request": "launch",
    "name": "Microchip Debug Adapter",
    "filePath": "${workspaceFolder}/build/src/my_app.elf",
    "device": "PIC24FJ128GA202",
    "toolType": "PICKit4",
    "toolOptions": { },
    "stopOnEntry": true
}
```

**Comment by callwyat** [comment]
Can you post the error message you are getting when trying to use the PICKit4? The toolOptions is know to cause issues with the 5 series of programmers. In a normal project, all the options are specified in the configuration.xml file. The one that I have the most trouble with is the power options. I was  able to test my bare bones project using the PICKit4 and empty tool options, but I may need to try a more complicated project.

**Comment by krister-ts** [comment]
After sitting awhile at `Progamming target...` I get the error "Failure to write program to device".

```shell
>Device PIC24FJ128GA202
HwTool PICKit4
>

 

*****************************************************


Connecting to MPLAB PICkit 4

Currently loaded versions:
Application version...........00.09.46
Boot version..................01.00.00
Script version................00.05.58
Script build number...........abbfe90327
Tool pack version ............1.15.1688
Target voltage detected
Target device PIC24FJ128GA202 found.
Device Revision Id  = 0x4
Device Id = 0x4c520000

>Program "C:\Users\user\Documents\projects\app/build/src/my_app.elf"
Programming target...
```
![image](https://github.com/callwyat/mplab-extension/assets/44512474/74c15a16-d6be-4e3d-a6ce-d8d21c8c21d9)

**Comment by callwyat** [comment]
Thanks for posting the debugger output! It looks like the slashes are changing directions in the file path. I would think that MDB would handle that gracefully, but just for a sanity check, can you update the launch.json filePath to use “\\” instead of “/“

**Comment by krister-ts** [comment]
> Thanks for posting the debugger output! It looks like the slashes are changing directions in the file path. I would think that MDB would handle that gracefully, but just for a sanity check, can you update the launch.json filePath to use “\\” instead of “/“

I thought of that too but no luck

**Comment by callwyat** [comment]
I just released version 0.1.20, and while this release doesn't directly address this issues, it may help. 0.1.20 fixes a problem VSCode would attempt to set breakpoints while programing the device, which may cause issue with programing. Could you two update the extension and let me know if this issue persists?

Thank you!

**Comment by callwyat** [comment]
It looks like #19 is similar to this one. The fix for that issue is being released in v0.1.22. I'm going to close this issue for now, but if the v0.1.22 doesn't fix the problem, feel free to open this issue again!
